consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala

```diff
@@ -27,8 +27,11 @@ trait BaseApi extends Closable {
     RetryPolicy.backoff(backoffs) {
       // We will assume 5xx are retryable, everything else is not for now
       case (_, Return(rep)) => rep.status.code >= 500 && rep.status.code < 600
+      // Looks like it could be either of 2 depending on where FutureCancelledException occurred
+      case (_, Throw(e: Failure)) if e.isFlagged(Failure.Interrupted) => false
+      case (_, Throw(e: FutureCancelledException)) => false
```

Comment:
Do you want me to update the way we cancel futures within this PR? Or should I just adjust this PR so retry filter will handle Interrupted failures properly and then submit another PR changing the way we canceling futures?